Doc/library/typing.rst

```diff
@@ -933,6 +933,55 @@ These can be used as types in annotations using ``[]``, each having a unique syn
 
    .. versionadded:: 3.9
 
+
+.. data:: TypeGuard
+
+   Special typing form used to annotate the return type of a user-defined
+   type guard function.  ``TypeGuard`` only accepts a single type argument.
+
+   ``TypeGuard`` aims to benefit *type narrowing* - a technique used by static
+   type checkers to determine a more precise type of an expression within a
+   program's code flow.  Usually type narrowing is done by analyzing
+   conditional code flow and applying the narrowing to a block of code.  The
+   conditional expression here is sometimes referred to as a "type guard".
+
+   Sometimes, a type guard can be a complex checking function.
+   These type guard functions require ``TypeGuard`` to narrow their input types
```

Comment:
I'd repeat a few times that we're talking about user-defined type guard functions here, not about the built-in functions such as `isinstance()` which are used to compose "built-in" type guards (not sure whether "built-in" is the right term actually).